AnaTutorial/src/AnaTutorial.cc

```diff
@@ -0,0 +1,849 @@
+{
+  if (Verbosity() > 1)
+  {
+    cout << "get the truth jets" << endl;
+  }
+
+  /// Get the truth jet node
+  JetMap *truth_jets = findNode::getClass<JetMap>(topNode, "AntiKt_Truth_r04");
+
+  if (!truth_jets)
+  {
+    cout << PHWHERE
+         << "Truth jet node is missing, can't collect truth jets"
+         << endl;
+    return;
+  }
+
+  /// Iterate over the truth jets
+  for (JetMap::Iter iter = truth_jets->begin();
+       iter != truth_jets->end();
+       ++iter)
+  {
+    const Jet *jet = iter->second;
+
+    m_truthjetpt = jet->get_pt();
+
+    /// Only collect truthjets above the _minjetpt cut
+    if (m_truthjetpt < m_minjetpt)
+      continue;
+
+    m_truthjeteta = jet->get_eta();
+    m_truthjetpx = jet->get_px();
+    m_truthjetpy = jet->get_py();
+    m_truthjetpz = jet->get_pz();
+    m_truthjetphi = jet->get_phi();
+    m_truthjetp = jet->get_p();
+    m_truthjetenergy = jet->get_e();
+
+    /// Fill the truthjet tree
+    m_truthjettree->Fill();
+  }
+}
+
+/**
+ * Get the reconstructed jets and store them in a tree
+ */
+void AnaTutorial::getReconstructedJets(PHCompositeNode *topNode)
+{
+  /// Get the reconstructed tower jets
+  JetMap *reco_jets = findNode::getClass<JetMap>(topNode, "AntiKt_Tower_r04");
+  /// Get the truth jets
+  JetMap *truth_jets = findNode::getClass<JetMap>(topNode, "AntiKt_Truth_r04");
+
+  if (!reco_jets)
+  {
+    cout << PHWHERE
+         << "Reconstructed jet node is missing, can't collect reconstructed jets"
+         << endl;
+    return;
+  }
+
+  if (Verbosity() > 1)
+  {
+    cout << "Get all Reco Jets" << endl;
+  }
+
+  /// Iterate over the reconstructed jets
+  for (JetMap::Iter recoIter = reco_jets->begin();
+       recoIter != reco_jets->end();
+       ++recoIter)
+  {
+    const Jet *jet = recoIter->second;
+    m_recojetpt = jet->get_pt();
+    if (m_recojetpt < m_minjetpt)
+      continue;
+
+    m_recojeteta = jet->get_eta();
+
+    // Get reco jet characteristics
+    m_recojetid = jet->get_id();
+    m_recojetpx = jet->get_px();
+    m_recojetpy = jet->get_py();
+    m_recojetpz = jet->get_pz();
+    m_recojetphi = jet->get_phi();
+    m_recojetp = jet->get_p();
+    m_recojetenergy = jet->get_e();
+
+    if (Verbosity() > 1)
+    {
+      cout << "matching by distance jet" << endl;
+    }
+
+    /// Set the matched truth jet characteristics to 0
+    m_truthjetid = 0;
+    m_truthjetp = 0;
+    m_truthjetphi = 0;
+    m_truthjeteta = 0;
+    m_truthjetpt = 0;
+    m_truthjetenergy = 0;
+    m_truthjetpx = 0;
+    m_truthjetpy = 0;
+    m_truthjetpz = 0;
+
+    /// Check to make sure the truth jet node is available
+    if (truth_jets)
+    {
+      /// Match the reconstructed jet to the closest truth jet in delta R space
+      /// Iterate over the truth jets
+      float closestjet = 9999;
+      for (JetMap::Iter truthIter = truth_jets->begin();
+           truthIter != truth_jets->end();
+           ++truthIter)
+      {
+        const Jet *jet = truthIter->second;
```

Comment:
suggest call this `truth_jet` instead of `jet`, as `jet` is used earlier on and can be confusing.

Reply:
Fixed